scrapy/pipelines/files.py

```diff
@@ -127,9 +128,22 @@ def _onsuccess(boto_key):
         return self._get_boto_key(path).addCallback(_onsuccess)
 
     def _get_boto_bucket(self):
-        # disable ssl (is_secure=False) because of this python bug:
-        # https://bugs.python.org/issue5103
-        c = self.S3Connection(self.AWS_ACCESS_KEY_ID, self.AWS_SECRET_ACCESS_KEY, is_secure=False)
+        # If using botocore, it is safe to use defaults.
+        if self.is_botocore():
+            c = self.S3Connection(self.AWS_ACCESS_KEY_ID, self.AWS_SECRET_ACCESS_KEY)
+        else:
+            # If the bucket name does not contain a dot,
+            # it is okay to use secure connections in boto.
+            if '.' not in self.bucket:
+                c = self.S3Connection(self.AWS_ACCESS_KEY_ID, self.AWS_SECRET_ACCESS_KEY)
+            # boto cannot handle secure connections properly if the bucket name contains dots
+            # https://github.com/boto/boto/issues/2836
+            # https://github.com/boto/boto/issues/443
+            else:
+                c = self.S3Connection(self.AWS_ACCESS_KEY_ID, self.AWS_SECRET_ACCESS_KEY, is_secure=False)
+                message = 'Insecure connection to S3 is used. Consider using botocore instead of boto.'
+                warnings.warn(message)
+
```

Comment:
What about this version:
```python
if not self.is_botocore() and '.' in self.bucket:
  # boto cannot handle secure connections properly if the bucket name contains dots
  # https://github.com/boto/boto/issues/2836
  # https://github.com/boto/boto/issues/443
  is_secure = False
  warnings.warn('Using insecure connection to S3. Consider using botocore instead of boto',
                category=ScrapyDeprecationWarning, stacklevel=1)
else:
  is_secure = True

c = self.S3Connection(self.AWS_ACCESS_KEY_ID, self.AWS_SECRET_ACCESS_KEY, is_secure=is_secure)
```